### DISCUSSION TITLE
splinepy cli

### DISCUSSION THREAD

**Discussion by j042** [tataratat/splinepy#258] [General]
Let's add some cli scripts to the package. To start with, a `load-and-show`. Please extend ideas here!

```
splinepy --show splinepy-spline.json
```

**Comment by j042** [comment]
@jzwar @danielwolff1 @clemens-fricke

**Comment by danielwolff1** [comment]
So the idea is to have a set of executable command-line interfaces/scripts that solve some tasks that are often necessary, like e.g., visualising splines, maybe converting between different formats of splines, etc...?

**Reply by j042** [reply]
Visualizing, yes. We can also do simple conversions, but there're different options for different IO types and it'd be unnecessarily complicated to support them. However, if you catch yourself opening python interactive mode just to do this one specific task very often and you can abstract general use-case of this situation, you should consider adding it

**Comment by clemens-fricke** [comment]
As far as I know you can also add multiple commands like splinepy-show and splinepy-solve or something like that. 

It would make the interfaces easier. 

I will look into it when I get home.

**Comment by clemens-fricke** [comment]
Soo I have made a spike with command stubs. These have to be correctly implemented now.
The current method is sub commands
splinepy convert -i ...
splinepy plot -i ...
![image](https://github.com/tataratat/splinepy/assets/3420870/ab80ffe3-7cea-436f-81e9-faa3f759d7e1)


It is also possible to do separate commands like 'splinepy-plot' and 'splinepy-convert'. What would you think is better?

**Reply by j042** [reply]
single command

**Reply by clemens-fricke** [reply]
I might have forgotten to mention that the subcommand method is implemented as a spike in the branch [splinepy:ft-cli](https://github.com/tataratat/splinepy/tree/ft-cli). Feel free to check it out. If I find the time I will also work on implementing the plotting function, but I would like to quickly talk about some things before I do it.

**Reply by jzwar** [reply]
I like the subcommands, but in general I am not really in favor of these scripts. It is just one more thing to maintain, also one which is very hard to test

**Reply by j042** [reply]
good point. Maybe only `show`?

**Reply by clemens-fricke** [reply]
With show what are the default options? And here we also have the problem with import options. Should it just import with default options and show it?

**Comment by j042** [comment]
@clemens-fricke can we have something like `splinepy --version`? that would show the versions of splinepy and its dependencies

**Comment by j042** [comment]
I am still interested in having `splinepy --show <file>`. Maybe we can support minimal set of options, such as control points, knots, resolutions, ...

**Reply by clemens-fricke** [reply]
#357
